hbase-server/src/main/java/org/apache/hadoop/hbase/util/TableDescriptorChecker.java

```diff
@@ -82,8 +83,11 @@ public static void sanityCheck(final Configuration c, final TableDescriptor td)
 
     // check max file size
     long maxFileSizeLowerLimit = 2 * 1024 * 1024L; // 2M is the default lower limit
-    long maxFileSize = td.getMaxFileSize();
-    if (maxFileSize < 0) {
+    String maxFileSizeValue = td.getValue(TableDescriptorBuilder.MAX_FILESIZE);
+    long maxFileSize = maxFileSizeValue == null? -1 : Long.parseLong(maxFileSizeValue);
+    // if not set MAX_FILESIZE in TableDescriptor,
+    // use maxFileSizeLowerLimit instead to skip this check
+    if (maxFileSizeValue == null) {
```

Comment:
Is this a little awkward? Should it be....

long maxFileSize = td.getMaxFileSize() == null? conf.getLong(HConstants.HREGION_MAX....): Long.parseLong(td.getMaxFileSize()); ?

Reply:
Thanks for the suggestion, fix it now.